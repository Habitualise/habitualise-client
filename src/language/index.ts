export const LABEL = {
  WELCOME_HEADER: 'Ready to\nbecome giga?',
  WELCOME_SUBHEADER: 'Start your journey with us\ntoday on project habits.',
  SIGN_UP: 'Sign Up',
  ALREADY_HAVE_ACCOUNT: 'I already have an account',
  LOGIN_SCREEN: 'Login Screen',
  LOG_IN: 'Log In',
  LOG_OUT: 'Log Out',
  WELCOME_SCREEN: 'Welcome Screen',
  TODAY_SCREEN: 'Today Screen',
  TODAY_HEADER: "Today's Habits",
  TODAY: 'Today',
  HABITS_SCREEN: 'Habits Screen',
  HABITS_HEADER: 'My Habits',
  ACTIVE: 'Active',
  ARCHIVED: 'Archived',
  HABITS: 'Habits',
  GROUPS_SCREEN: 'Groups Screen',
  GROUPS: 'Groups',
  SETTINGS_SCREEN: 'Settings Screen',
  SETTINGS: 'Settings',
  LOGGED_IN_AS: (username: string) => `Logged in as ${username}`,
  DAYS_COMPLETED: (percentage: number) => `${percentage}% of days completed`,
  ADD_HABIT: 'Add a new habit',
  HABIT_NAME: 'Habit Name',
  REPEAT: 'REPEAT',
};
