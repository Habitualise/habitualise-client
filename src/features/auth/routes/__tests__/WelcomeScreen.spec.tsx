import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';

import { WelcomeScreen } from '../WelcomeScreen';
import { LABEL } from '@/language';

const mockAuthorize = jest.fn();

jest.mock('react-native-auth0', () => ({
  useAuth0: () => ({
    authorize: mockAuthorize,
  }),
}));

describe('WelcomeScreen', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<WelcomeScreen />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should call useAuth0 authorize function when login button is pressed', () => {
    render(<WelcomeScreen />);

    fireEvent.press(screen.getByText(LABEL.SIGN_UP));
    expect(mockAuthorize).toHaveBeenCalled();
  });
});
