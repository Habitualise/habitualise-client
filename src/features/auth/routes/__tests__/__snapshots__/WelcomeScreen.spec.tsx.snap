// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(255, 251, 254, 1)",
        },
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 15,
        },
      ]
    }
  >
    <Image
      source={
        {
          "testUri": "../../../apps/mobile/src/app/assets/googleLogo.png",
        }
      }
      style={
        {
          "height": 150,
          "marginBottom": 50,
          "resizeMode": "contain",
          "width": 150,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontSize": 45,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 52,
          },
          {
            "textAlign": "left",
          },
          {
            "writingDirection": "ltr",
          },
          {
            "color": "rgb(0, 99, 154)",
            "fontWeight": "700",
            "marginBottom": 15,
            "textAlign": "center",
          },
        ]
      }
    >
      Ready to
become giga?
    </Text>
    <Text
      style={
        [
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          {
            "textAlign": "left",
          },
          {
            "writingDirection": "ltr",
          },
          {
            "color": "rgb(0, 99, 154)",
            "lineHeight": 30,
            "marginBottom": 60,
            "textAlign": "center",
          },
        ]
      }
    >
      Start your journey with us
today on project habits.
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgb(0, 99, 154)",
          },
          {
            "borderRadius": 10,
            "marginBottom": 10,
            "paddingHorizontal": 120,
            "paddingVertical": 10,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0.15,
              "lineHeight": 24,
            },
            {
              "textAlign": "left",
            },
            {
              "writingDirection": "ltr",
            },
            {
              "color": "white",
              "fontWeight": "500",
            },
          ]
        }
      >
        Sign Up
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        {
          "alignSelf": undefined,
          "bottom": undefined,
          "end": undefined,
          "left": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "start": undefined,
          "top": undefined,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                false,
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  {
                    "flexDirection": "row-reverse",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "marginLeft": -4,
                      "marginRight": 12,
                    },
                    {
                      "marginLeft": -16,
                      "marginRight": 16,
                    },
                    {
                      "marginLeft": -8,
                      "marginRight": 12,
                    },
                  ]
                }
                testID="button-icon-container"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontSize": 20,
                      },
                      undefined,
                      {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  î—ˆ
                </Text>
              </View>
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 16,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                I already have an account
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
